File: components/http-request/src/main/kotlin/io/terminus/dalaran/support/component/http/client/HttpClientConfig.java
package io.terminus.dalaran.support.component.http.client;

import io.terminus.dalaran.annotation.ConfigFieldInfo;
import io.terminus.dalaran.config.FieldInputType;
import lombok.Data;

@Data
public class HttpClientConfig {
    @ConfigFieldInfo(label = "协议", inputType = FieldInputType.Radio)
    private HttpProtocol protocol;
    private String host;
    private Integer port;
    private String path;
    private HttpMethod method;
}

File: components/http-request/src/main/kotlin/io/terminus/dalaran/component/http/request/HttpClientConfig.kt
package io.terminus.dalaran.component.http.request

import io.terminus.dalaran.annotation.DalaranConfigField
import io.terminus.dalaran.annotation.FieldInputType

class HttpClientConfig(
        @DalaranConfigField(label = "协议", inputType = FieldInputType.Radio)
        val protocol: HttpRequestProtocol,
        val host: String,
        val port: String,
        val path: String,
        val method: String
)